pkg/plugins/policies/meshratelimit/plugin/v1alpha1/plugin_test.go

```diff
@@ -0,0 +1,458 @@
+package v1alpha1_test
+
+import (
+	"path/filepath"
+	"time"
+
+	envoy_resource "github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"google.golang.org/protobuf/types/known/durationpb"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	core_plugins "github.com/kumahq/kuma/pkg/core/plugins"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	core_xds "github.com/kumahq/kuma/pkg/core/xds"
+	api "github.com/kumahq/kuma/pkg/plugins/policies/meshratelimit/api/v1alpha1"
+	plugin "github.com/kumahq/kuma/pkg/plugins/policies/meshratelimit/plugin/v1alpha1"
+	policies_xds "github.com/kumahq/kuma/pkg/plugins/policies/xds"
+	gateway_plugin "github.com/kumahq/kuma/pkg/plugins/runtime/gateway"
+	test_matchers "github.com/kumahq/kuma/pkg/test/matchers"
+	test_model "github.com/kumahq/kuma/pkg/test/resources/model"
+	test_xds "github.com/kumahq/kuma/pkg/test/xds"
+	util_proto "github.com/kumahq/kuma/pkg/util/proto"
+	xds_context "github.com/kumahq/kuma/pkg/xds/context"
+	envoy_common "github.com/kumahq/kuma/pkg/xds/envoy"
+	. "github.com/kumahq/kuma/pkg/xds/envoy/listeners"
+	"github.com/kumahq/kuma/pkg/xds/generator"
+)
+
+var _ = Describe("MeshRateLimit", func() {
+	type sidecarTestCase struct {
+		resources            []*core_xds.Resource
+		fromRules            core_xds.FromRules
+		inboundRateLimitsMap core_xds.InboundRateLimitsMap
+		expectedListeners    []string
+	}
+	DescribeTable("should generate proper Envoy config",
+		func(given sidecarTestCase) {
+			resourceSet := core_xds.NewResourceSet()
+			resourceSet.Add(given.resources...)
+
+			context := xds_context.Context{
+				Mesh: xds_context.MeshContext{
+					Resource: &core_mesh.MeshResource{
+						Meta: &test_model.ResourceMeta{
+							Name: "default",
+						},
+					},
+				},
+			}
+			proxy := core_xds.Proxy{
+				Dataplane: &core_mesh.DataplaneResource{
```

Comment:
you should use dataplane builder from https://github.com/kumahq/kuma/blob/52c992533bb8ab5f565e56f1b7a9af0a96e361a7/pkg/test/resources/builders/dataplane_builder.go#L18